components/pages/dashboard/ads/AdsForm.tsx

```diff
@@ -0,0 +1,319 @@
+
+type FormValue = {
+  adName: string
+  startDate: string
+  endDate: string
+  image: any
+  headline: string
+  description: string
+  landingUrl: string
+  callToActionUrl: string
+  discordPlacement: string
+  telegramPlacement: string
+  metaversePlacement: string
+}
+
+export function AdsForm() {
+  const { back } = useRouter()
+  const { handleSubmit, control, watch } = useForm<FormValue>({
+    defaultValues: {
+      discordPlacement: '',
+    },
+  })
+
+  const startDate = watch('startDate')
+
+  const onSubmit = (values: FormValue) => {
+    console.log(values)
+  }
+
+  const discordOptions = [
+    {
+      src: '/assets/ads/discord-1546x423.png',
+      label: '1546 x 423',
+      value: '1',
+    },
+    {
+      src: '/assets/ads/512x512.png',
+      label: '512 x 512',
+      value: '2',
+    },
+    {
+      src: '/assets/ads/no-img.png',
+      label: 'No image',
+      value: '3',
+    },
+  ]
+
+  const telegramOptions = [
+    {
+      src: '/assets/ads/tele-1546x423.png',
+      label: '1546 x 423',
+      value: '1',
+    },
+    {
+      src: '/assets/ads/512x512.png',
+      label: '512 x 512',
+      value: '2',
+    },
+    {
+      src: '/assets/ads/no-img.png',
+      label: 'No image',
+      value: '3',
+    },
+  ]
+
+  const metaverseOptions = [
+    {
+      src: '/assets/ads/big-ad.png',
+      label: '1546 x 423',
+      value: '1',
+    },
+    {
+      src: '/assets/ads/small-ad.png',
+      label: '512 x 512',
+      value: '2',
+    },
+    {
+      src: '',
+      label: '',
+      value: '',
+    },
+  ]
+
+  const radioOptionClassName =
+    'flex flex-col w-fit p-[10px] gap-y-[10px] rounded-lg border-solid border-[3px] border-transparent cursor-pointer'
+
+  const renderCustomRadioGroup = ({
+    options,
+    optionClassName,
+  }: {
+    options: Record<string, any>[]
+    optionClassName: string
+  }) => {
+    return (
+      <RadioGroup
+        options={options.map(({ label, value }) => ({
+          label,
+          value,
+        }))}
+        radioGroupClassName="flex flex-row justify-between"
+        renderOption={(option, selectedValue) => {
+          return (
+            <div
+              className={clsx(radioOptionClassName, {
+                ' !border-mochi':
+                  selectedValue && option.value === selectedValue,
+              })}
+            >
+              <div className={optionClassName}>
+                {option.value && (
+                  <Image
+                    fill
+                    className="object-contain"
+                    alt={option.value}
+                    src={
+                      options.find((item) => item.value === option.value)
+                        ?.src || ''
+                    }
+                  />
+                )}
+              </div>
+              {!!option.label && <div className="text-sm">{option.label}</div>}
+            </div>
+          )
+        }}
+      />
+    )
+  }
```

Comment:
put this outside the component to save performance

Reply:
thanks, did it